packages/synapse-react-client/test/components/download_list/TableQueryDownloadConfirmation.test.tsx

```diff
@@ -0,0 +1,204 @@
+import { QueryVisualizationContextProvider } from '../../../src/components/QueryVisualizationWrapper'
+import { mockQueryBundleRequest } from '../../../mocks/mockFileViewQuery'
+import { cloneDeep } from 'lodash-es'
+import {
+  useAddQueryToDownloadList,
+  useGetQueryResultBundleWithAsyncStatus,
+} from '../../../src/synapse-queries'
+import {
+  getUseMutationMock,
+  getUseQuerySuccessMock,
+} from '../../testutils/ReactQueryMockUtils'
+
+jest.mock('../../../src/synapse-queries', () => ({
+  useGetQueryResultBundleWithAsyncStatus: jest.fn(),
+  useAddQueryToDownloadList: jest.fn(),
+}))
+
+const mockUseGetQueryResultBundle = jest.mocked(
+  useGetQueryResultBundleWithAsyncStatus,
+)
+const mockUseAddQueryToDownloadList = jest.mocked(useAddQueryToDownloadList)
+
+const DOWNLOAD_CONFIRMATION_UI_TEST_ID = 'DownloadConfirmationUI'
+const mockDownloadConfirmationUi = jest
+  .spyOn(DownloadConfirmationUIModule, 'DownloadConfirmationUI')
+  .mockImplementation(() => (
+    <div data-testid={DOWNLOAD_CONFIRMATION_UI_TEST_ID}></div>
+  ))
+
+const mockToastFn = jest
+  .spyOn(ToastMessage, 'displayToast')
+  .mockImplementation(() => {})
+
+const addFilesToDownloadListResponse: AddToDownloadListResponse = {
+  concreteType:
+    'org.sagebionetworks.repo.model.download.AddToDownloadListResponse',
+  numberOfFilesAdded: 1,
+}
+
+const mockSetShowDownloadConfirmation = jest.fn()
+
+async function setUp() {
+  const user = userEvent.setup()
+  let component
+  await act(() => {
+    component = render(
+      <QueryContextProvider
+        queryContext={{
+          getLastQueryRequest: () => cloneDeep(mockQueryBundleRequest),
+        }}
+      >
+        <QueryVisualizationContextProvider
+          queryVisualizationContext={{
+            setShowDownloadConfirmation: mockSetShowDownloadConfirmation,
+          }}
+        >
+          <TableQueryDownloadConfirmation />
+        </QueryVisualizationContextProvider>
+      </QueryContextProvider>,
+    )
+  })
+
+  await screen.findByTestId(DOWNLOAD_CONFIRMATION_UI_TEST_ID)
+  return { component, user }
+}
+
+describe('TableQueryDownloadConfirmation', () => {
+  let mutationMockReturnValue: ReturnType<
+    typeof getUseMutationMock<
+      AddToDownloadListResponse,
+      SynapseClientError,
+      AddToDownloadListRequest
+    >
+  >
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    mockUseGetQueryResultBundle.mockReturnValue(
+      getUseQuerySuccessMock<
+        AsynchronousJobStatus<QueryBundleRequest, QueryResultBundle>
+      >({
+        responseBody: {
+          queryCount: 100,
+          sumFileSizes: {
+            sumFileSizesBytes: 40128868,
+            greaterThan: false,
+          },
+        },
+      }),
+    )
+
+    mutationMockReturnValue = getUseMutationMock<
+      AddToDownloadListResponse,
+      SynapseClientError,
+      AddToDownloadListRequest
+    >(addFilesToDownloadListResponse)
+
+    mockUseAddQueryToDownloadList.mockReturnValue(mutationMockReturnValue)
```

Comment:
Starting to think instead of mocking `SynapseClient`, we create reusable utilities to mock our query/cache layer.

Ideally we will also have unit tests for the query/cache layer that validate the right APIs are called, and that the right cache keys are invalidated.

Reply:
That works.  At one point I had the idea that all APIs would be mocked, but that was so painful (and potentially contaminating).  This seems like the right layer to mock